src/main/webapp/app/exercises/programming/manage/programming-exercise-detail.component.html

```diff
@@ -235,23 +235,23 @@ <h2><span jhiTranslate="artemisApp.programmingExercise.detail.title">Programming
                         </jhi-clone-repo-button>
                         <jhi-programming-exercise-instructor-repo-download
                             class="ml-2"
-                            [exerciseId]="programmingExercise.id"
+                            [exerciseId]="programmingExercise.id!"
                             [repositoryType]="'TESTS'"
                         ></jhi-programming-exercise-instructor-repo-download>
                     </div>
                 </dd>
                 <dt><span jhiTranslate="artemisApp.programmingExercise.templateBuildPlanId">Template Build Plan Id</span></dt>
                 <dd>
                     <span *ngIf="programmingExercise.templateParticipation && programmingExercise.templateParticipation.buildPlanId"
-                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.templateParticipation.buildPlanId" [projectKey]="programmingExercise.projectKey">{{
+                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.templateParticipation.buildPlanId" [projectKey]="programmingExercise.projectKey!">{{
                             programmingExercise.templateParticipation.buildPlanId
                         }}</a></span
                     >
                 </dd>
                 <dt><span jhiTranslate="artemisApp.programmingExercise.solutionBuildPlanId">Solution Build Plan Id</span></dt>
                 <dd>
                     <span *ngIf="programmingExercise.solutionParticipation && programmingExercise.solutionParticipation.buildPlanId"
-                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.solutionParticipation.buildPlanId" [projectKey]="programmingExercise.projectKey">{{
+                        ><a jhiBuildPlanLink [buildPlanId]="programmingExercise.solutionParticipation.buildPlanId!" [projectKey]="programmingExercise.projectKey!">{{
```

Comment:
any reason we need a `!` after buildplan here but not above in the templateParticipation? For both it is defined as:
`buildPlanId?: string`

Reply:
Good question. I'll remove it and see if the build fails

Reply:
Indeed it doesn't cause a problem so I reverted that change.